gammapy/maps/core.py

```diff
@@ -894,11 +894,120 @@ def interp_to_geom(self, geom, preserve_counts=False, fill_value=0, **kwargs):
+        method : {'oversampling', 'polygon'}
+            Method to reproject data to a new projection
+            -oversampling: the base map is oversampled and 
+             its bins are re-distributed into a map with the target geom.
+            -polygon: flux-conserving spherical polygon intersection
+            from reproject.reproject_exact
+        oversampling_factor : int
+           Minimal factor between the bin size of the output map and the oversampled base map.
+           Used only for the oversampling method.
+
+        Returns
+        -------
+        output_map : `Map`
+            Reprojected Map
+        """
+        from .region import RegionGeom
+
+        if isinstance(geom, RegionGeom):
+            raise TypeError(f"Reproject is not supported for {type(geom)}")
+        if isinstance(self.geom, RegionGeom):
+            raise TypeError(f"Reproject is not supported for {type(self.geom)}")
+        
+        if geom.is_image:
+            axes = [ax.copy() for ax in self.geom.axes]
+            geom = geom.copy(axes=axes)
+            axes_eq = True
+        elif geom.ndim != self.geom.ndim:
+            raise TypeError("Invalid axes in the target geom.")
+        else:
+            axes_eq = np.all(
+                [ax0 == ax1 for ax0, ax1 in zip(geom.axes, self.geom.axes)]
+            )
+
+        if not axes_eq:
+            target_geom = geom.copy(axes=self.geom.axes) 
+            requires_interp = True
+        else:
+            target_geom = geom
+            requires_interp = False
+        
+        if method == "oversampling":
+            new_map = self._reproject_oversampling(target_geom,
+                                                   preserve_counts=preserve_counts,
+                                                   oversampling_factor=oversampling_factor)
+        elif method == "polygon":
+            new_map = self._reproject_polygon(target_geom,
+                                              preserve_counts=preserve_counts)
+        else:
+            raise TypeError("Available methods are 'oversampling' or 'polygon'.")
+
+        if requires_interp:
+            return new_map.interp_to_geom(geom, preserve_counts=preserve_counts, fill_value=fill_value)
+        else:
+            return new_map
+
+    def _reproject_oversampling(self, geom, preserve_counts=False, oversampling_factor=10):
+        output_map = Map.from_geom(geom, unit=self.unit)
+        base_factor = geom.pixel_scales.min()/self.geom.pixel_scales.min()
+        if base_factor >= oversampling_factor:
+            input_map = self
+        else:
+            factor = int(np.ceil(oversampling_factor / base_factor))
+            input_map = self.upsample(factor=factor, preserve_counts=preserve_counts)
+
+        coords = input_map.geom.get_coord()
+        output_map.fill_by_coord(coords,
+                                 weights=input_map.quantity,
+                                 preserve_counts=preserve_counts
```

Comment:
`fill_by_coord` makes sense only for maps of integral quantities but the method exits independently of the type of data, so I think it is useful to have the option preserve_counts=False to do the average over the filled bins

Reply:
we can't simply use the oversampling factor because  the weight is not necessarily the same for all the bins as we may have less pixels that contribute at the edges